src/opm/input/eclipse/Schedule/Well/WellConnections.cpp

```diff
@@ -461,6 +466,20 @@ namespace Opm {
             double re = std::sqrt(D[0] * D[1] / angle * 2); // area equivalent radius of the grid block
             double connection_length = D[2];            // the length of the well perforation
 
+            // adapt connection_length for non default CF or Kh
+            if (Kh > 0 || CF > 0 ) {
+                std::array<double,3> cell_perm = {{ props->permx,
+                                                    props->permy,
+                                                    props->permz}};
+                const auto& K = permComponents(direction, cell_perm);
+                double Ke = std::sqrt(K[0] * K[1]);
+                double KhEff = Kh;
+                if (CF > 0 && Kh < 0) {
+                    KhEff = CF * (std::log(r0 / std::min(r0, rw)) + skin_factor) / angle;
+                }
+                connection_length = KhEff / Ke;
+            }
+
```

Comment:
This will affect the polymer model and the filer cake model. If this change is not wanted for polymer and the filter cake model I can instead add Ke as computed from the grid as a member in the connection  @GitPaean Any commets?

Reply:
After some more thinking I decided to add Ke as suggested above. Relaying on a modified connection_length is not a good solution in the long term.